bin/vault/usage.py

```diff
@@ -229,7 +275,21 @@ def _resolve_path(path: T.Path) -> T.Path:
 
             # Resolve all paths
             parsed.files = [_resolve_path(path) for path in parsed.files]
-
+        
+        if "fofn" in parsed and parsed.fofn is not None:
+
+            def _create_fofn_generator(fofn: T.Path) -> T.Iterator[T.Path]:
+                # Check: Any problems with yield from within a context manager?   
+                with open(fofn) as file:
+                    while file_line := file.readline():
+                        filepath = T.Path(file_line.rstrip())
+                        resolved_path = filepath.resolve()
+                        if T.Path(filepath).is_symlink():
+                            log.warning(f"{filepath} is a symlink. Acting on the original file: {resolved_path}")
+                        yield resolved_path
+
+            parsed.files = _create_fofn_generator(parsed.fofn)
```

Comment:
`parsed.files` here is a generator, whilst on above (`parsed.files = [_resolve_path(path) for path in parsed.files]`) it is a list - yes, iterating over them is the same, but it would be very good to decide whether `parsed.files` is a list or generator

Reply:
possibly also we should make use of the `_resolve_path` function again, as we are doing the same thing. Maybe move the reading the file above `# Resolve all paths`, and then have the same line do the resolving symlinks, no matter whether they came from the fofn or the args

Reply:
`parsed.files` is supposed to be an `iterable`: thats the only functionality we use. So it can either be a list or generator or another iterable. I disagree that it would be good to decide on one - that it is a list or a generator. 

The type annotation originally had it to be a `list`. I've changed it to `iterable` (loading large fofns into a list will take up memory and might crash).

Reply:
agree about `_resolve_path` - I've made the change